src/components/Debugger/DebuggerModal.tsx

```diff
@@ -0,0 +1,151 @@
+import { ChromeUser } from '@redhat-cloud-services/types';
+import { useDispatch, useSelector } from 'react-redux';
+import { DeepRequired } from 'utility-types';
+
+import { toggleDebuggerModal } from '../../redux/actions';
+import { ReduxState } from '../../redux/store';
+import { Select, SelectList, SelectOption } from '@patternfly/react-core/next';
+import { MenuToggle, MenuToggleElement } from '@patternfly/react-core';
+
+import './Debugger.scss';
+
+export type DebuggerModalProps = {
+  user: DeepRequired<ChromeUser>;
+};
+
+const DebuggerModal = ({ user }: DebuggerModalProps) => {
+  const isOpen = useSelector<ReduxState, boolean | undefined>(({ chrome: { isDebuggerModalOpen } }) => isDebuggerModalOpen);
+  const dispatch = useDispatch();
+  const setIsModalOpen = (isOpen: boolean) => dispatch(toggleDebuggerModal(isOpen));
+  const handleCloseModal = () => {
+    setIsModalOpen(false);
+  };
+  const [isDropdownOpen, setIsOpen] = React.useState(false);
+  const [selected, setSelected] = React.useState<string>('Entitlements');
+  const menuRef = React.useRef<HTMLDivElement>(null);
+  const onToggleClick = () => {
+    setIsOpen(!isDropdownOpen);
+  };
+  const onSelect = (_event: React.MouseEvent<Element, MouseEvent> | undefined, itemId: string | number | undefined) => {
+    setSelected(itemId as string);
+    setIsOpen(false);
+  };
+  const toggle = (toggleRef: React.Ref<MenuToggleElement>) => (
+    <MenuToggle ref={toggleRef} onClick={onToggleClick} isExpanded={isDropdownOpen} className="chr-c-debugger-dropdown">
+      {selected}
+    </MenuToggle>
+  );
+
+  const entitlements = Object.entries(user.entitlements).reduce(
+    (acc, [key, entitlement]) => ({
+      ...acc,
+      [`entitlements_${key}`]: entitlement.is_entitled,
+      [`entitlements_${key}_trial`]: entitlement.is_trial,
+    }),
+    {}
+  );
+  const transformEntitlements = Object.entries(entitlements).map((val, index) => {
+    //remove entitlement if false
+    if (val[1] === true) {
+      return (
+        <tr key={index}>
+          <td>{val[0].replace(/entitlements_/g, '')}</td>
+        </tr>
+      );
+    } else {
+      return null;
+    }
+  });
+
+  const [permissions, setPermissions] = React.useState({});
+  React.useEffect(() => {
+    async function getUserPermissions() {
+      const userPermissions = await window.insights.chrome.getUserPermissions();
+      const userPermissionsList = Object.entries(userPermissions).reduce(
+        (acc, [key, userPermission]) => ({
+          ...acc,
+          [`userPermissionsList_${key}`]: userPermission.permission,
+        }),
+        {}
+      );
+      setPermissions(userPermissionsList);
+    }
+    getUserPermissions();
+  }, []);
+  const transformPermissions = Object.values(permissions).map((val, index) => {
+    return (
+      <tr key={index}>
+        <td>{val}</td>
+      </tr>
+    );
+  });
+
+  function renderSelection() {
```

Comment:
Can you make this a component instead of scoped function? it's generally not a good practice to create these render functions instead of components.